Spigot-Server-Patches/0562-Climbing-should-not-bypass-cramming-gamerule.patch

```diff
@@ -0,0 +1,179 @@
+     }
+ 
+     @Override
+-    public boolean isCollidable() {
++    public boolean isCollidable(boolean ignoreClimbing) { // Paper
+         return false;
+     }
+ 
+diff --git a/src/main/java/net/minecraft/server/EntityBoat.java b/src/main/java/net/minecraft/server/EntityBoat.java
+index 7df2837d6a058d375e8d45b7c00fac343abaf7fa..ce2145e314218697b06f02572c329305d6a7a8c6 100644
+--- a/src/main/java/net/minecraft/server/EntityBoat.java
++++ b/src/main/java/net/minecraft/server/EntityBoat.java
+@@ -106,7 +106,7 @@ public class EntityBoat extends Entity {
+     }
+ 
+     @Override
+-    public boolean isCollidable() {
++    public boolean isCollidable(boolean ignoreClimbing) { // Paper
+         return true;
+     }
+ 
+diff --git a/src/main/java/net/minecraft/server/EntityHorseAbstract.java b/src/main/java/net/minecraft/server/EntityHorseAbstract.java
+index 8921c2333cd5efcb87dd52b95fe12fd5340df55f..cc298f28d376022ef34e14183425c86e66261e5e 100644
+--- a/src/main/java/net/minecraft/server/EntityHorseAbstract.java
++++ b/src/main/java/net/minecraft/server/EntityHorseAbstract.java
+@@ -165,7 +165,7 @@ public abstract class EntityHorseAbstract extends EntityAnimal implements IInven
+     }
+ 
+     @Override
+-    public boolean isCollidable() {
++    public boolean isCollidable(boolean ignoreClimbing) { // Paper
+         return !this.isVehicle();
+     }
+ 
+diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
+index b92e74349b402b6a8687876073f974128fe242a4..2662e5b1c4d47e5c76e39f0c19e41404483d31c0 100644
+--- a/src/main/java/net/minecraft/server/EntityLiving.java
++++ b/src/main/java/net/minecraft/server/EntityLiving.java
+@@ -42,7 +42,6 @@ import org.bukkit.event.entity.EntityTeleportEvent;
+ import org.bukkit.event.player.PlayerItemConsumeEvent;
+ // CraftBukkit end
+ 
+-import co.aikar.timings.MinecraftTimings; // Paper
+ 
+ public abstract class EntityLiving extends Entity {
+ 
+@@ -2835,7 +2834,7 @@ public abstract class EntityLiving extends Entity {
+             return;
+         }
+         // Paper - end don't run getEntities if we're not going to use its result
+-        List<Entity> list = this.world.getEntities(this, this.getBoundingBox(), IEntitySelector.a(this));
++        List<Entity> list = this.world.getEntities(this, this.getBoundingBox(), IEntitySelector.pushable(this, world.paperConfig.fixClimbingBypassingCrammingRule)); // Paper - fix climbing bypassing cramming rule
+ 
+         if (!list.isEmpty()) {
+             // Paper - move up
+@@ -2973,9 +2972,16 @@ public abstract class EntityLiving extends Entity {
+         return !this.dead && this.collides; // CraftBukkit
+     }
+ 
+     @Override
+     public boolean isCollidable() {
++        // Paper start
+-        return this.isAlive() && !this.isSpectator() && !this.isClimbing() && this.collides; // CraftBukkit
++        return this.isCollidable(world.paperConfig.fixClimbingBypassingCrammingRule);
```

Comment:
This wasnt neccessary since `Entity#isCollidable()` already forwards to `isCollidable(boolean)`

Reply:
It was when testing. Without that one, e.g. pigs in vines would not cram to death even though there were well over 50 pigs there; the second I started the server with, they crammed to death.

Reply:
You likely didn't add the param to this method in your test? My patch made it so the only `isCollidable()` was in `Entity`

Will test here in a minute and see whats up

Reply:
Ok, properly tested and confirmed you did fix a bug with my impl here. Carry on ^_^